### PR TITLE
BDD: Apply Best Promotion scenario

### PR DESCRIPTION
**Pull request by sukumarh**
Finished the UI changes and BDD scenario for Apply Best Promotion.

### CODE REVIEW

**Review by nspektor — COMMENTED** [review]
Looks pretty good! some notes:
- Make sure to also add product-pricings to the clear function
- when manually following the steps laid out in your behave scenario, I got the following results, including Promo8 twice, is that expected? it makes sense to me that you could apply Promo8 to both products but just wanted to make sure. 
Screen Shot 2020-11-19 at 10 00 17 AM

**Comment by sukumarh** [comment]
> Looks pretty good! some notes:
> 
> * Make sure to also add product-pricings to the clear function
> * there are some merge conflicts to fix
> * when manually following the steps laid out in your behave scenario, I got the following results, including Promo8 twice, is that expected? it makes sense to me that you could apply Promo8 to both products but just wanted to make sure.
>   Screen Shot 2020-11-19 at 10 00 17 AM

Thanks! I finished the following.
1. Clear function also clears the product pricing.
2. Fixed the merge conflicts with _master_.
3. Refactored some of the nested code (displaying results) into another function.
4. Changed the comment to **Apply Best Promotion for the products**.

About the results, there were two ways to proceed. Let's assume the results are as in the screenshot. I went for the second one, since it is more of a suggestion for each product. What do you suggest?
1. Products grouped together:

| PromoCode  | Products        |
| -------------- | -------------- |
| Promo8         | 256,269         |

2. One row per product:

| PromoCode  | Products        |
| -------------- | -------------- |
| Promo8         | 256                |
| Promo8         | 269                |

File: service/static/js/rest_api.js [hunk 1]
```diff
@@ -345,4 +345,91 @@ $(function () {
         });
 
     });
+
+    // ****************************************
+    // Apply Best Promotionfor a Promotion
```
[the remark this answers is not in the document]
Comment (sukumarh) [reply service/static/js/rest_api.js]: Yes! Thanks, fixed it in the last merge.

**Comment by nspektor** [comment]
Screen Shot 2020-11-20 at 11 36 49 AM
I am getting this error when I run honcho start, then run behave, then manually type 101,123,256,269 into "Products Associated with Promotion:" and press "Apply best promotion"
any idea whats going on?

**Comment by sukumarh** [comment]
> Screen Shot 2020-11-20 at 11 36 49 AM
> 
> I am getting this error when I run honcho start, then run behave, then manually type 101,123,256,269 into "Products Associated with Promotion:" and press "Apply best promotion" any idea whats going on? also some new merge conflicts

I tried the same, this time I destroyed the VM and re-created it, and it worked. Going to look into the merge conflicts and let you know once I'm done.

image

**Comment by sukumarh** [comment]
@nspektor I finished the merge with the master. All the steps should be working now.

**Comment by liondon** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/82ed68f3-a995-419e-80a4-f9191b98d0a4?env_id=ibm:yp:us-south "nyu-devops-promo-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/a90cf2f8-b961-4e4c-98ff-49891084d12b "nyu-devops-promo-pipeline") deployed [promotions](https://github.com/2020-nyu-devops/promotions/releases/tag/deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201202-221756 "deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201202-221756") to [dev](https://console.bluemix.net/apps/42c5c4c9-a7f2-472a-bfb1-add0f77b8a15?env_id=ibm:yp:us-south "ibm:yp:us-south:2020-nyu-devops:dev"), including commit 959a952c5d83968db97dc26d035e43baebe33d0d

**Pull request merged**